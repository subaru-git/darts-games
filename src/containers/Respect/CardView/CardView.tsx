import React, { FC, Fragment } from 'react';
import { ExternalLinkIcon } from '@chakra-ui/icons';
import { Box, Center, Flex, Heading, Link, Image, Text } from '@chakra-ui/react';
import replace from 'react-string-replace';
import Card from '@/components/Card';

type CardViewProps = {
  data: RespectResult;
};

const CardView: FC<CardViewProps> = ({ data }) => {
  return (
    <Flex direction='column' p={4}>
      <Box>
        <Heading as='h2' pb={4}>
          Companies
        </Heading>
        <Flex wrap={'wrap'} gap={2}>
          {data.companies.map((item) => (
            <div key={item.title}>
              <Card>
                <CardContent data={item} />
              </Card>
            </div>
          ))}
        </Flex>
      </Box>
      <Box pt={4}>
        <Heading as='h2' pb={4}>
          Professional
        </Heading>
        <Flex wrap={'wrap'} gap={2}>
          {data.professionals.map((item) => (
            <div key={item.title}>
              <Card>
                <CardContent data={item} />
              </Card>
            </div>
          ))}
        </Flex>
      </Box>
      <Box pt={4}>
        <Heading as='h2' pb={4}>
          Youtube
        </Heading>
        <Flex wrap={'wrap'} gap={2}>
          {data.youtube.map((item) => (
            <div key={item.title}>
              <Card>
                <CardContent data={item} />
              </Card>
            </div>
          ))}
        </Flex>
      </Box>
    </Flex>
  );
};

const CardContent: FC<{ data: WebsiteResult }> = ({ data }) => {
  return (
    <>
      <Center>
        <Link href={data.url ?? '#'} isExternal>
          <Image
            src={data.image ?? ''}
            alt={`${data.title} image`}
            fallbackSrc='/fallback.webp'
            height='150px'
            maxWidth='400px'
          />
        </Link>
      </Center>
      <Heading as='h3' size={{ base: 'md', md: 'lg' }} py={2}>
        {data.title}
      </Heading>
      <Text fontSize={{ base: 'sm', md: 'md' }} wordBreak='break-word'>
        {replaceDescription(data.description ?? '')}
      </Text>
    </>
  );
};

const replaceDescription = (description: string) => {
  const urlReplaced = replace(
    description,
    /(https?:\/\/[\w/:%#\$&\?\(\)~\.=\+\-]+)/g,
    (match, i) => {
      return (
        <Fragment key={i}>
          <br />
          <Link href={match} isExternal color='teal.700' fontStyle='italic'>
            {match} <ExternalLinkIcon />
          </Link>
          <br />
        </Fragment>
      );
    },
  );
  return urlReplaced;
};

export default CardView;
