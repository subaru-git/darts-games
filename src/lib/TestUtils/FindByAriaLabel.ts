export const findByAriaLabel = (elements: HTMLElement[], label: string) => {
  return (
    elements.find((button) => button.attributes.getNamedItem('aria-label')?.value === label) ??
    new HTMLElement()
  );
};
