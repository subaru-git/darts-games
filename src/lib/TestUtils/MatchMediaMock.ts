import mediaQuery from 'css-mediaquery'

export const createMatchMedia =
  (width: number) =>
  (query: string): MediaQueryList => ({
    matches: mediaQuery.match(query, { width }),
    media: query,
    onchange: null,
    addListener: jest.fn(), // deprecated
    removeListener: jest.fn(), // deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })
