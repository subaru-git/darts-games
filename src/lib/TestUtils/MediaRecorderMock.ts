export const MediaRecorderMock = {
  start: jest.fn(),
  ondataavailable: jest.fn(),
  onerror: jest.fn(),
  state: '',
  stop: jest.fn(),
};
