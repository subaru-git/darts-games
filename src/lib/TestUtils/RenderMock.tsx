import { ReactNode } from 'react';
import { ChakraProvider } from '@chakra-ui/react';
import { render as defaultRender } from '@testing-library/react';
import Providers from '@/contexts/Providers';

const render = (children: ReactNode | ReactNode[]) =>
  defaultRender(
    <ChakraProvider>
      <Providers>{children}</Providers>
    </ChakraProvider>,
  );
export { render };
