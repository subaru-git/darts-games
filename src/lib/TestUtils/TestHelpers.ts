import { waitFor as originalWaitFor, waitForOptions } from '@testing-library/react';

const defaultTimeout = process.env.WAIT_FOR_DEFAULT_TIMEOUT ? parseInt(process.env.WAIT_FOR_DEFAULT_TIMEOUT, 10) : 5000;

export async function waitFor(callback: () => void,  options?: waitForOptions): Promise<void> {
  return originalWaitFor(callback, { timeout: defaultTimeout, ...options });
}