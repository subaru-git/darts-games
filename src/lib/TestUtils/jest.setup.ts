import { createMatchMedia } from '@/lib/TestUtils/MatchMediaMock';
import { MediaRecorderMock } from '@/lib/TestUtils/MediaRecorderMock';

jest.mock('next/router', () => ({ useRouter: jest.fn().mockReturnValue({ locale: 'en' }) }));
jest.mock('react-camera-pro');

beforeAll(() => {
  window.matchMedia = createMatchMedia(1100);
  window.MediaRecorder = (jest.fn() as any).mockImplementation(() => MediaRecorderMock);
  global.ResizeObserver = require('resize-observer-polyfill');
});
