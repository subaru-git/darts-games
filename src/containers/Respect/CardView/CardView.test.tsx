import CardView from './CardView';
import { render } from '@/lib/TestUtils/RenderMock';

test('should rendering', () => {
  const { container } = render(
    <CardView
      data={{
        companies: [
          {
            title: 'title',
            image: 'image',
            description: 'description',
            url: 'url',
            type: 'type',
          },
        ],
        professionals: [],
        youtube: [],
      }}
    />,
  );
  expect(container).toMatchSnapshot();
});
