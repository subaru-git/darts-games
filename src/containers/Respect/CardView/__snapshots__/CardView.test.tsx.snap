// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should rendering 1`] = `
<div>
  <div
    class="css-15wiczr"
  >
    <div
      class="css-0"
    >
      <h2
        class="chakra-heading css-1uy00aa"
      >
        Companies
      </h2>
      <div
        class="css-18z2gjh"
      >
        <div>
          <div
            class="css-18xfez7"
          >
            <div
              class="css-gmuwbf"
            >
              <a
                class="chakra-link css-spn4bz"
                href="url"
                rel="noopener"
                target="_blank"
              >
                <img
                  alt="title image"
                  class="chakra-image__placeholder css-1l0a0ng"
                  src="/fallback.webp"
                />
              </a>
            </div>
            <h3
              class="chakra-heading css-1mbs55a"
            >
              title
            </h3>
            <p
              class="chakra-text css-1ykf6wi"
            >
              description
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="css-1rwovhe"
    >
      <h2
        class="chakra-heading css-1uy00aa"
      >
        Professional
      </h2>
      <div
        class="css-18z2gjh"
      />
    </div>
    <div
      class="css-1rwovhe"
    >
      <h2
        class="chakra-heading css-1uy00aa"
      >
        Youtube
      </h2>
      <div
        class="css-18z2gjh"
      />
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
